core/src/main/java/org/axonframework/messaging/annotation/MultiHandlerDefinition.java

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright (c) 2010-2016. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.messaging.annotation;
+
+import org.axonframework.common.annotation.PriorityAnnotationComparator;
+
+import java.lang.reflect.Executable;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * HandlerDefinition instance that delegates to multiple other instances, in the order provided. Also, it wraps the
+ * delegate with {@link HandlerEnhancerDefinition}.
+ *
+ * @author Tyler Thrailkill
+ * @author Milan Savic
+ * @since 3.3
+ */
+public class MultiHandlerDefinition implements HandlerDefinition {
```

Comment:
I'm assuming this guy is the reason why the `HandlerEnhancerDefinition` isn't pulled out through the `ServiceLoader` in the `AnnotatedHandlerInspector#initializeMessageHandlers()` function. Is that correct?

Reply:
Yes, we're combining here creation and wrapping of a handler.